openedx/core/djangoapps/course_groups/partition_generator.py

```diff
@@ -0,0 +1,72 @@
+"""
+The team dynamic partition generation to be part of the
+openedx.dynamic_partition plugin.
+"""
```

Comment:
The `course_groups` djangoapp is essentially a plugin interface, these team partition generator is just one plugin based on that interface. So, in order to keep the interface decoupled from its plugins, I think we need to move this module out of the `course_groups` djangoapp.

One place this code could fit is the existing openedx/djangoapps/teams_config.py module. What do you think @mariajgrimaldi ?